sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/security/CertificateNamesGenerator.java

```diff
@@ -0,0 +1,64 @@
+package com.mesosphere.sdk.offer.evaluate.security;
+
+import com.mesosphere.sdk.api.EndpointUtils;
+import org.bouncycastle.asn1.x500.X500Name;
+import org.bouncycastle.asn1.x509.GeneralName;
+import org.bouncycastle.asn1.x509.GeneralNames;
+
+/**
+ * A {@link CertificateNamesGenerator} creates relevant names for given service pod.
+ */
+public class CertificateNamesGenerator {
+
+    private String serviceName;
+    private String taskName;
+
+    public CertificateNamesGenerator(String serviceName, String taskName) {
+        this.serviceName = serviceName;
+        this.taskName = taskName;
+    }
+
+    /**
+     * Returns a Subject for service certificate.
+     * @return
+     */
+    public X500Name getSubject() {
+       // TODO(mh): Should we somehow improve certificate name here? Right now it reflects only service
```

Comment:
What are valid values for the common name? Any reason to not shove everything in?